example-deploy.py

```diff
@@ -20,21 +20,60 @@ def run_and_capture(*argv):
     return subprocess.check_output(argv)
 
 
-def build(*components):
+def synchronize(*components):
+    """Synchronize the code repositories with upstreams.
+
+    This is called once on the code host with a list of all components to
+    synchronize. The script should fetch from upstream remotes, sync with a SCM
+    server, or whatever your system does. The command should return a mapping
+    of each component to an object containing a unique identifier for this
+    version of the code and optionally the hostname of a buildhost.
+
+    If a buildhost is provided for a component, rollingpin will run the build
+    command on the build host with a list of all components destined for it.
+
+    If no buildhost is provided, the token returned by synchronize will be
+    passed straight through to the deploy command on each host.
+
+    This command is required if you want to run "-d" commands in rollingpin.
+
+    """
+
+    return {
+        # a component that needs to be built
+        "buildable": {
+            "token": "7be0db612ea365e1d9410763198bb79a9e28dfd6",
```

Comment:
After looking at https://github.com/reddit/reddit-private/pull/760/files I think I gathered it - the token is just the commit sha that the synchronize fetched up to. I think I was thrown off by example-deploy and thought it would be a fixed token as some sort of identifier of the type of component being built or something.